components/Dialogs/tests/snapshot/MDCAlertControllerDynamicTypeTests.m

```diff
@@ -0,0 +1,280 @@
+ */
+- (void)setAlertControllerTraitCollectionSizeToSize:(UIContentSizeCategory)sizeCategory {
+  UITraitCollection *traitCollection = [[UITraitCollection alloc] init];
+  if (@available(iOS 10.0, *)) {
+    traitCollection =
+        [UITraitCollection traitCollectionWithPreferredContentSizeCategory:sizeCategory];
+  }
+
+  self.alertController.traitCollectionOverride = traitCollection;
+}
+
+/** Test when a @c MDCAlertController has a content size of @c UIContentSizeCategorySmall. */
+- (void)testContentSizeCategorySmall {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategorySmall];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+/** Test when a @c MDCAlertController has a content size of @c UIContentSizeCategoryMedium. */
+- (void)testContentSizeCategoryMedium {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryMedium];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+/** Test when a @c MDCAlertController has a content size of @c UIContentSizeCategoryLarge. */
+- (void)testContentSizeCategoryLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+/** Test when a @c MDCAlertController has a content size of @c UIContentSizeCategoryExtraLarge. */
+- (void)testContentSizeCategoryExtraLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryExtraLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+/** Test when a @c MDCAlertController has a content size of @c UIContentSizeCategoryExtraExtraLarge.
+ */
+- (void)testContentSizeCategoryExtraExtraLarge {
+  // Given
+  [self setAlertControllerTraitCollectionSizeToSize:UIContentSizeCategoryExtraExtraLarge];
+
+  // When
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+/** Test when a @c MDCAlertController has a content size of @c
+ * UIContentSizeCategoryExtraExtraExtraLarge. */
+- (void)testContentSizeCategoryExtraExtraExtraLarge {
```

Comment:
Please name these with something that reflects their Given/When values. Perhaps:

```
// Material Typography-based DT
testTypographyDynamicTypeForContentSizeCategoryExtraExtraLarge
testTypographyDynamicTypeForContentSizeCategoryExtraExtraExtraLarge

// Scalable font-based DT
testScaledFontDynamicTypeForContentSizeCategoryExtraExtraLarge
testScaledFontDynamicTypeForContentSizeCategoryExtraExtraExtraLarge
```

Reply:
Done